From 6e1a436f4f11817143df7445a7a0e89a171066d5 Mon Sep 17 00:00:00 2001
From: imkenf <imkenf@gmail.com>
Date: Fri, 13 Jun 2025 20:18:44 +0800
Subject: [PATCH] fridmira: Add FridMira framework to base library build system

---
 lib/base/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/base/meson.build b/lib/base/meson.build
index 0eab53f..4f1680d 100644
--- a/lib/base/meson.build
+++ b/lib/base/meson.build
@@ -3,6 +3,7 @@ base_sources = [
   'session-glue.c',
   'channel.vala',
   'service.vala',
+  'fridmira.vala',
   'stream.vala',
   'rpc.vala',
   'p2p.vala',
-- 
2.45.1.windows.1

