From 43517d802bc6f27dabb4c37d72cbe15125ff2b08 Mon Sep 17 00:00:00 2001
From: imkenf <imkenf@gmail.com>
Date: Thu, 12 Jun 2025 16:22:18 +0800
Subject: [PATCH 2/6] fridmira: Integrate FridMira module into build system

---
 src/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/meson.build b/src/meson.build
index 34a212a..37e9f0b 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -1,6 +1,7 @@
 base_sources = [
   'frida.vala',
   'frida-glue.c',
+  'fridmira.vala',
   'control-service.vala',
   'portal-service.vala',
   'host-session-service.vala',
-- 
2.45.1.windows.1

